src/utilities/graphql/print.ts

```diff
@@ -1,23 +1,23 @@
 import type { ASTNode } from "graphql";
 import { print as origPrint } from "graphql";
-import { canUseWeakMap } from "../common/canUse.js";
+import { WeakCache } from "@wry/caches";
 
-let printCache: undefined | WeakMap<ASTNode, string>;
-// further TODO: replace with `optimism` with a `WeakCache` once those are available
+let printCache!: WeakCache<ASTNode, string>;
 export const print = Object.assign(
   (ast: ASTNode) => {
-    let result;
-    result = printCache?.get(ast);
+    let result = printCache.get(ast);
 
     if (!result) {
-      result = origPrint(ast);
-      printCache?.set(ast, result);
+      printCache.set(ast, (result = origPrint(ast)));
```

Comment:
I the assignment and passing of the value can be done on a single line, but I find this style so hard to read. In fact, I almost started a new review comment here pointing out that you missed the `result` assignment before realizing it was assigned in the parameter.

Would it be possible to leave the assignment separate from the value passed to the argument as you had it before? Its much easier to follow.

```suggestion
      result = origPrint(ast)
      printCache.set(ast, result);
```

Reply:
I assumed it would add to the size, but funny enough, changing to your suggestion adds one byte to `dist/apollo-client.min.cjs` and removes one byte from `dist/index.js`, so it's net-neutral or even better for what we hope most of our users use.
Making the change :)